src/librustc_trans/debuginfo/mod.rs

```diff
@@ -428,7 +428,7 @@ pub fn create_function_debug_context<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
             // If the method does *not* belong to a trait, proceed
             if cx.tcx().trait_id_of_impl(impl_def_id).is_none() {
                 let impl_self_ty =
-                    common::def_ty(cx.tcx(), impl_def_id, instance.substs);
+                    Instance::new(impl_def_id, instance.substs).ty(cx.tcx());
```

Comment:
This shouldn't have been using `def_ty` at all... But `cx.tcx().type_of(impl_def_id).subst(cx.tcx(), instance.substs)`.

Reply:
But that should be the same as `Instance::new(impl_def_id, instance.substs).ty(cx.tcx());`?

Reply:
The error comes from https://github.com/MaikKlein/rust/blob/collector/src/librustc_trans/trans_item.rs#L174

Reply:
Wait sorry it actually comes from `type_of`,

```
  10: rustc_trans::type_of::in_memory_type_of
  11: rustc_trans::trans_item::predefine_static
  12: rustc_trans::trans_item::MonoItemExt::predefine
  13: rustc_trans::base::compile_codegen_unit
  14: rustc::ty::maps::<impl rustc::ty::maps::queries::compile_codegen_unit<'tcx>>::compute_result
  15: rustc::dep_graph::graph::DepGraph::with_task_impl
  16: rustc::dep_graph::graph::DepGraph::with_task
  17: rustc_errors::Handler::track_diagnostics
  18: rustc::ty::maps::plumbing::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::cycle_check
  19: rustc::ty::maps::<impl rustc::ty::maps::queries::compile_codegen_unit<'tcx>>::force
  20: rustc::ty::maps::<impl rustc::ty::maps::queries::compile_codegen_unit<'tcx>>::try_get
  21: rustc::ty::maps::TyCtxtAt::compile_codegen_unit
  22: rustc::ty::maps::<impl rustc::ty::context::TyCtxt<'a, 'tcx, 'lcx>>::compile_codegen_unit
  23: rustc_trans::base::trans_crate
```

Reply:
Does making *this* specific change cause an error? It's unclear to me, you seem to have replied pointing to a different change.